### ISSUE TITLE
Setting device rules for cgroups v1 doesn't work - "Transition" function's delta is always blank

### ISSUE THREAD

**Issue by iholder-redhat** [opencontainers/runc#3141]
Hey,

I'm a Kubevirt developer.
For a while now we do the following to apply device rules:
```
target, err := loadEmulator(path)
...
target.Apply(*rule)
```

This is because [Transition function](https://github.com/opencontainers/runc/blob/master/libcontainer/cgroups/fs/devices.go#L74)  that is supposed to "Compute the minimal set of transition rules needed to achieve the requested state", as written in the comments, always returns an empty delta although it's clear that such a delta should exist.

Tried on v1.0.0.

**Comment by kolyshkin** [comment]
@iholder-redhat I'm afraid your bug report is not very clear. Can you please describe what are you doing, what you think should happen, and what actually happens? Or, provide a simple repro.

**Comment by kolyshkin** [comment]
@iholder-redhat also please take a look at [devices_emulator_test.go](https://github.com/opencontainers/runc/blob/master/libcontainer/cgroups/devices/devices_emulator_test.go) -- you can find test cases in which Transition() do not return empty rule set.

Ideally, demonstrate a bug by adding a test case to the above file.

**Comment by kolyshkin** [comment]
@iholder-redhat ^^^^

**Comment by iholder-redhat** [comment]
> @iholder-redhat ^^^^

Hey @kolyshkin,
It's just been a while since I've encountered this bug. Next time I will save all info & way to reproduce when I encounter it so the issue will be more valuable - sorry for that. I thought it's better to have it here without good info than forget about it. In any case I've tried to reproduce now and I see that the issue is gone. That's good news - I assume that I was working with an older version back then and it was fixed meanwhile.

We had that workaround in Kubevirt for a while, so good news :)
